x-pack/plugins/security_solution/server/endpoint/services/actions/clients/crowdstrike/crowdstrike_actions_client.ts

```diff
@@ -120,34 +120,33 @@ export class CrowdstrikeActionsClient extends ResponseActionsClientImpl {
   }
 
   private async getEventDetailsById(agentId: string): Promise<{
-    crowdstrike: { event: { HostName: string } };
+    host: { name: string };
   }> {
     const search = {
       index: ['logs-crowdstrike.fdr*', 'logs-crowdstrike.falcon*'],
       size: 1,
-      _source: ['crowdstrike.event.HostName'],
+      _source: ['host.name'],
       body: {
         query: {
           bool: {
-            filter: [{ term: { 'crowdstrike.event.DeviceId': agentId } }],
+            filter: [{ term: { 'device.id': agentId } }],
           },
         },
       },
     };
-    const result: SearchResponse<{ crowdstrike: { event: { HostName: string } } }> =
-      await this.options.esClient
-        .search<{ crowdstrike: { event: { HostName: string } } }>(search, {
-          ignore: [404],
-        })
-        .catch((err) => {
-          throw new ResponseActionsClientError(
-            `Failed to fetch event document: ${err.message}`,
-            err.statusCode ?? 500,
-            err
-          );
-        });
-
-    return result.hits.hits?.[0]?._source as { crowdstrike: { event: { HostName: string } } };
+    const result: SearchResponse<{ host: { name: string } }> = await this.options.esClient
+      .search<{ host: { name: string } }>(search, {
+        ignore: [404],
+      })
+      .catch((err) => {
+        throw new ResponseActionsClientError(
+          `Failed to fetch event document: ${err.message}`,
+          err.statusCode ?? 500,
+          err
+        );
+      });
+
+    return result.hits.hits?.[0]?._source as { host: { name: string } };
```

Comment:
The way you are casting there with no prior checks can lead to bugs.
Can we improve the runtime checks here instead of casting?  Example: check that the results has at least one hit and that hit has `host.name`... if not then `throw`.

Reply:
Good point, thank you 👍